src/Services/CarePlanService.php

```diff
@@ -0,0 +1,276 @@
+use OpenEMR\Services\Search\ReferenceSearchField;
+use OpenEMR\Services\Search\ReferenceSearchValue;
+use OpenEMR\Services\Search\SearchModifier;
+use OpenEMR\Services\Search\StringSearchField;
+use OpenEMR\Services\Search\TokenSearchField;
+use OpenEMR\Services\Search\TokenSearchValue;
+use OpenEMR\Validators\ProcessingResult;
+use Twig\Token;
+
+class CarePlanService extends BaseService
+{
+    const SURROGATE_KEY_SEPARATOR = "_";
+    private const PATIENT_TABLE = "patient_data";
+    private const ENCOUNTER_TABLE = "form_encounter";
+    private const CARE_PLAN_TABLE = "form_care_plan";
+
+    const TYPE_PLAN_OF_CARE = 'plan_of_care';
+    const TYPE_GOAL = 'goal';
+
+    const CARE_PLAN_TYPES = [self::TYPE_PLAN_OF_CARE, self::TYPE_GOAL];
+
+    /**
+     * @var string
+     */
+    private $carePlanType;
+
+    /**
+     * @var CodeTypesService
+     */
+    private $codeTypesService;
+
+    function getUuidFields(): array
+    {
+        return ['puuid', 'euuid'];
+    }
+
+    public function __construct($carePlanType = self::TYPE_PLAN_OF_CARE)
+    {
+        if (in_array($carePlanType, self::CARE_PLAN_TYPES) !== false) {
+            $this->carePlanType = $carePlanType;
+        } else {
+            throw new \InvalidArgumentException("Invalid care plan type of " . $carePlanType);
+        }
+
+        (new UuidRegistry(['table_name' => self::PATIENT_TABLE]))->createMissingUuids();
+        (new UuidRegistry(['table_name' => self::ENCOUNTER_TABLE]))->createMissingUuids();
+
+        parent::__construct(self::CARE_PLAN_TABLE);
+        $this->codeTypesService = new CodeTypesService();
+    }
+
+    public function getOne($uuid, $puuid = null)
+    {
+        $search = [
+            'uuid' => new TokenSearchField('uuid', [new TokenSearchValue($uuid, null, false)])
+        ];
+        if (isset($puuid)) {
+            $search['puuid'] = new ReferenceSearchField('puuid', [new ReferenceSearchValue($puuid, 'Patient', true)]);
+        }
+        return $this->search($search);
+    }
+
+    /**
+     * Returns a list of all care plan resources.  Search array can be a simple key => value array which does an exact
+     * match on passed in value.  For more complicated searching @see CarePlanService::search().
+     * @param $search a key => value array
+     * @param bool $isAndCondition Whether the search should be a UNION of search values or INTERSECTION of search values
+     * @return ProcessingResult
+     */
+    public function getAll($search, $isAndCondition = true)
+    {
+        $newSearch = [];
+        foreach ($search as $key => $value) {
+            if (!$value instanceof ISearchField) {
+                $newSearch[] = new StringSearchField($key, [$value], SearchModifier::EXACT);
+            } else {
+                $newSearch[$key] = $value;
+            }
+        }
+        return $this->search($search, $isAndCondition);
+    }
+
+    public function search($search, $isAndCondition = true)
```

Comment:
does there need to be support for a `$puuid = null` parameter?

Reply:
actually, looks like it is usually on the getAll request.

Reply:
This should be fixed now.